src/YDialogSpy.cc

```diff
@@ -138,93 +209,211 @@ YDialogSpy::YDialogSpy( YDialog * targetDialog )
+    add_items.push_back( m1 );
+    add_items.push_back( m2 );
+    add_items.push_back( m3 );
+    add_items.push_back( m4 );
+    add_items.push_back( m5 );
+    add_items.push_back( m6 );
+    add_items.push_back( m7 );
+
+    new YMenuItem( m1, "Label" );
+    new YMenuItem( m1, "Heading" );
+    new YMenuItem( m1, "RichText" );
+    new YMenuItem( m1, "ProgressBar" );
+    new YMenuItem( m1, "BusyIndicator" );
+    new YMenuItem( m1, "Table" );
+
+    new YMenuItem( m2, "PushButton" );
+    new YMenuItem( m2, "CheckBox" );
+    new YMenuItem( m2, "ComboBox" );
+    new YMenuItem( m2, "MenuButton" );
+    new YMenuItem( m2, "RadioButton" );
+
+    new YMenuItem( m3, "InputField" );
+    new YMenuItem( m3, "IntField" );
+    new YMenuItem( m3, "MultiLineEdit" );
+    new YMenuItem( m3, "MultiSelectionBox" );
+    new YMenuItem( m3, "Password" );
+    new YMenuItem( m3, "SelectionBox" );
+
+    new YMenuItem( m4, "Left" );
+    new YMenuItem( m4, "Right" );
+    new YMenuItem( m4, "Top" );
+    new YMenuItem( m4, "Bottom" );
+    new YMenuItem( m4, "HCenter" );
+    new YMenuItem( m4, "VCenter" );
+    new YMenuItem( m4, "HVCenter" );
+
+    new YMenuItem( m5, "MinHeight" );
+    new YMenuItem( m5, "MinWidth" );
+    new YMenuItem( m5, "MinSize" );
+    new YMenuItem( m5, "HSquash" );
+    new YMenuItem( m5, "VSquash" );
+    new YMenuItem( m5, "HVSquash" );
+    new YMenuItem( m5, "HWeight" );
+    new YMenuItem( m5, "VWeight" );
+
+    new YMenuItem( m6, "MarginBox" );
+    new YMenuItem( m6, "ButtonBox" );
+    new YMenuItem( m6, "CheckBoxFrame" );
+    new YMenuItem( m6, "Frame" );
+    new YMenuItem( m6, "HBox" );
+    new YMenuItem( m6, "HSpacing" );
+    new YMenuItem( m6, "ReplacePoint" );
+    new YMenuItem( m6, "VBox" );
+    new YMenuItem( m6, "VSpacing" );
+
+    // TODO: these are not available in ncurses UI
+    new YMenuItem( m7, "BarGraph" );
+    new YMenuItem( m7, "DateField" );
+    new YMenuItem( m7, "DumbTab" );
+    new YMenuItem( m7, "Graph" );
+    new YMenuItem( m7, "Slider" );
+    new YMenuItem( m3, "TimeField" );
+    new YMenuItem( m7, "TimezoneSelector" );
+
+    priv->addButton->addItems( add_items );
+
+    priv->deleteButton = fac->createPushButton( hbox, "&Delete" );
+    priv->upButton = fac->createPushButton( hbox, "⬆ Up" );
+    priv->downButton = fac->createPushButton( hbox, "⬇ Down" );
 
-    YAlignment * alignment = fac->createLeft( vbox );
-    priv->propButton       = fac->createPushButton( alignment, "&Properties >>>" );
     priv->propReplacePoint = fac->createReplacePoint( vbox );
     fac->createEmpty( priv->propReplacePoint );
 
-    YButtonBox * buttonBox = fac->createButtonBox( vbox );
-    priv->closeButton      = fac->createPushButton( buttonBox, "&Close" );
-    priv->closeButton->setRole( YOKButton );
+    priv->selectedWidgetChanged();
 }
 
-
+/**
+ * Destructor
+ */
 YDialogSpy::~YDialogSpy()
 {
     if ( priv->spyDialog )
 	priv->spyDialog->destroy();
```

Comment:
That seems to provide just some simple check before calling the destructor (https://github.com/libyui/libyui/blob/interactive_ui_editor/src/YDialog.cc#L252), but don't ask me for details, I'm not the author.. :wink: